src/RTTComms.cpp

```diff
@@ -537,12 +540,20 @@ namespace BrainCloud
             _eventQueueMutex.lock();
             _callbackEventQueue.push_back(RTTCallback(RTTCallbackType::ConnectSuccess));
             _eventQueueMutex.unlock();
+            }
+            else if (operation == "DISCONNECT")
+            {
+                _disconnectedWithReason = true; // RESET THIS VALUE IN CLEANUP/connect()/ETC. Very error prone addition. be careful, have me review
+                _disconnectReasonMessage = json["data"]["status_message"].asString(); // Check log figure out what the message is
```

Comment:
We need to find a way to repro this, because I'm not even certain the json will look like that. @gmowhiz can confirm

Reply:
The format you will receive is:
`{
   "service": "rtt",
   "operation": "DISCONNECT",
   "data": {
      "reason": "Some text telling you why",
      "reasonCode": 40303
}
`